### PR TITLE
Added KHR_lights_punctual gltf extension

### PR DESCRIPTION
**Pull request by tlf30**
This PR adds support for the GLTF KHR_lights_punctual extension which allows lights from gltf files to be imported.
https://github.com/KhronosGroup/glTF/tree/master/extensions/2.0/Khronos/KHR_lights_punctual

Please feel free to give feedback. I am currently looking for advice on implementing the intensity field from the gltf spec into the jme lights. 

GLTF model taken from the KHR_light_punctual repo listed above.

### CODE REVIEW

**Comment by tlf30** [comment]
Still having an issue with rotation on spot lights.

**Comment by Markil3** [comment]
I have a fix for spotlight rotations, as I detail [here](https://hub.jmonkeyengine.org/t/gltf-unlit-material-and-punctual-light-extensions/44132/39?u=markil3). However, the fix is useful enough that I've suggested we may move it to a seperate PR.

**Comment by tlf30** [comment]
Currently blocked on #1466

**Comment by tlf30** [comment]
#1466 was merged, I will update this ASAP.

**Comment by tlf30** [comment]
OK, I think this is ready for review.

File: jme3-plugins/src/gltf/java/com/jme3/scene/plugins/gltf/LightsPunctualExtensionLoader.java [hunk 1]
```diff
@@ -0,0 +1,252 @@
...
+
+        //Spot specific
+        JsonObject spot = obj.getAsJsonObject("spot");
+        float innerConeAngle = spot != null && spot.has("innerConeAngle") ? spot.get("innerConeAngle").getAsFloat() : 0f;
+        float outerConeAngle = spot != null && spot.has("outerConeAngle") ? spot.get("outerConeAngle").getAsFloat() : ((float) Math.PI) / 4f;
+
+        /*
+        Correct floating point error on half PI, GLTF spec says that the outerConeAngle
+        can be less or equal to PI/2, but JME requires less than PI/2.
+        We will being the angle within PI/2 if it is equal or larger than PI/2
```
Comment (stephengold) [inline jme3-plugins/src/gltf/java/com/jme3/scene/plugins/gltf/LightsPunctualExtensionLoader.java]: being -> bring

Comment (tlf30) [reply jme3-plugins/src/gltf/java/com/jme3/scene/plugins/gltf/LightsPunctualExtensionLoader.java]: Fixed

File: jme3-plugins/src/gltf/java/com/jme3/scene/plugins/gltf/LightsPunctualExtensionLoader.java [hunk 2]
```diff
@@ -0,0 +1,295 @@
...
+        color.g = val;
+        color.b = val;
+
+        return color;
+    }
+
+    /**
+     * A bean to contain the relation between a node and a light index
+     */
+    private static class NodeNeedingLight {
```
Comment (stephengold) [inline jme3-plugins/src/gltf/java/com/jme3/scene/plugins/gltf/LightsPunctualExtensionLoader.java:270]: Does this private class need public methods?

Comment (tlf30) [reply jme3-plugins/src/gltf/java/com/jme3/scene/plugins/gltf/LightsPunctualExtensionLoader.java:270]: No, good catch. I will correct when you are finished with your review.

**Pull request merged**